### DISCUSSION TITLE
NetBox v3.2-beta1 is available for testing!

### DISCUSSION THREAD

**Discussion by jeremystretch** [netbox-community/netbox#8652] [Announcements]
I'm happy to announce that the first beta release for NetBox v3.2 is [now available for testing](https://github.com/netbox-community/netbox/releases/tag/v3.2-beta1)! Please bear in mind that this is a pre-release and thus **not suitable for production use**.

While we're a bit early with this beta - v3.2.0 is scheduled for April - we wanted to get the new plugins framework in front of plugin developer as early as possible to maximize their opportunity to provide feedback ahead of the final release. And there's _quite_ a lot to test, including new support for

* Generic views & corresponding templates
* Model features (change logging, custom fields, journaling, webhooks, etc.)
* Base classes for forms, tables, and filter sets
* Various custom template tags & filters

Our hope is that this release greatly enhances the plugin development experience, and reduces the overall burden for plugin maintainers.

Of course, it's not just about plugins: v3.2 introduces a ton of new core features as well, such as:

* Plugins Framework Extensions (#8333)
* Modules & Module Types (#7844)
* Custom Object Fields (#7006)
* Custom Status Choices (#8054)
* Improved User Preferences (#7759)
* Inventory Item Roles (#3087)
* Inventory Item Templates (#8118)
* Service Templates (#1591)
* Automatic Provisioning of Next Available VLANs (#2658)

...and a bunch more.

The complete releases notes for this pre-release are available [on ReadTheDocs](https://netbox.readthedocs.io/en/feature/release-notes/version-3.2/), as is the documentation for the [newly extended plugins framework](https://netbox.readthedocs.io/en/feature/plugins/development/). We've also extended NetBox's [device type library](https://github.com/netbox-community/devicetype-library) to start hosting module type definitions as well, so be sure to check that out as well!

As is tradition, we have spun up a dedicated demo instance for the beta release at https://beta-demo.netbox.dev/. The login credentials are `admin`:`admin` (just as with the regular demo). Please help us kick the tires! Your feedback will help to ensure a solid v3.2.0 release in April.

**Comment by raschkeb** [comment]
Hi, 

not sure if I should post here or open a bug: 

On the beta-demo site, when entering any integer value in the search field, I get a crash message from Django:

<class 'django.core.exceptions.FieldError'>

Cannot resolve keyword 'asn' into field. Choices are: _name, asns, circuit_terminations, clusters, comments, contacts, created, custom_field_data, description, devices, facility, group, group_id, id, images, journal_entries, last_updated, latitude, locations, longitude, name, physical_address, powerpanel, prefixes, racks, region, region_id, shipping_address, slug, status, tagged_items, tags, tenant, tenant_id, time_zone, vlan_groups, vlans

Python version: 3.8.10
NetBox version: 3.2.0-beta1


Strings or prefixes seem to work fine.

Cheers,
    Bernd

**Reply by compgeniuses** [reply]
How i wish this Release Candidate is available in Release soonest. i wish its released 1st March. Would be Fantastic.
Though i am curios on Migration path from previous, is it as tedious, or straightforward and utomated.

**Comment by jeremystretch** [comment]
FYI [beta2](https://github.com/netbox-community/netbox/releases/tag/v3.2-beta2) is now available. We plan to release v3.2.0 the week of April 4th.

**Reply by compgeniuses** [reply]
Fantastic, Only patience can help  i see, but its commendable Job on the new version. alot of things i know went into planning, thought and implementation, as well as migrations.
